packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -173,27 +180,47 @@ class DriveCommand extends RunCommandBase {
       'VM_SERVICE_URL': observatoryUri,
     };
 
-    sync_io.WebDriver driver;
+    async_io.WebDriver driver;
     // For web device, WebDriver session will be launched beforehand
     // so that FlutterDriver can reuse it.
     if (isWebPlatform) {
+      final Browser browser = _browserNameToEnum(
+          argResults['browser-name'].toString());
+      final String driverPort = argResults['driver-port'].toString();
       // start WebDriver
-      final Browser browser = _browserNameToEnum(argResults['browser-name'].toString());
-      driver = _createDriver(
-        argResults['driver-port'].toString(),
-        browser,
-        argResults['headless'].toString() == 'true',
-      );
+      try {
+        driver = await _createDriver(
+          driverPort,
+          browser,
+          argResults['headless'].toString() == 'true',
+        );
+      } catch (ex) {
+        throwToolExit(
+          'Unable to start WebDriver Session for Flutter for Web testing. \n'
+          'Make sure you have the correct WebDriver Server running at $driverPort. \n'
+          'Make sure the WebDriver Server matches option browser-name. \n'
+          '$ex'
+        );
+      }
 
       // set window size
       final List<String> dimensions = argResults['browser-dimension'].split(',') as List<String>;
       assert(dimensions.length == 2);
-      final int x = int.parse(dimensions[0]);
-      final int y = int.parse(dimensions[1]);
-      final sync_io.Window window = driver.window;
+      int x, y;
       try {
-        window.setLocation(const math.Point<int>(0, 0));
-        window.setSize(math.Rectangle<int>(0, 0, x, y));
+        x = int.parse(dimensions[0]);
+        y = int.parse(dimensions[1]);
+      } on FormatException catch (ex) {
+        throw FormatException('''
+        Dimension provided is invalid.
```

Comment:
This should probably be a tool exit. If you use `'''` then you will pick up all the whitespace to the left, instead format it like: 

```
        throwToolExit('''
Dimension provided to --browser-dimension is invalid:
$ex
''');
```

Reply:
Done.